icu4c/source/common/uresbund.cpp

```diff
@@ -1910,10 +1910,74 @@ static Resource getTableItemByKeyPath(const ResourceData *pResData, Resource tab
   return resource;
 }
 
-U_CAPI UResourceBundle* U_EXPORT2 
-ures_getByKeyWithFallback(const UResourceBundle *resB, 
-                          const char* inKey, 
-                          UResourceBundle *fillIn, 
+static void createPath(CharString& path,
+                       const char* resPath,
+                       int32_t     resPathLen,
+                       const char* origResPath,
+                       int32_t     origResPathLen,
+                       const char* inKey,
+                       UErrorCode* status) {
```

Comment:
nit/optional (that is, you can ignore me):
I would order the args like this:
1. origResPath (base)
2. resPath (input 1)
3. inKey (input 2)
4. path (output)
5. status (UErrorCode last)

Reply:
Eh, okay.